analytics/management/commands/create_ga_report.py

```diff
@@ -0,0 +1,325 @@
+                            help='The data dimensions to use when querying for specified metrics, such as page path')
+
+        parser.add_argument('--metrics',
+                            default='ga:pageviews,ga:uniquePageviews,ga:avgTimeOnPage',
+                            dest='metrics',
+                            help='The aggregated statistics for user activity to your site, such as clicks or pageviews')
+
+        # TODO: This doesn't actually make sense; the pivot logic is dependent on the other dimensions and the other
+        # filters...maybe have an optional parameter that you specify like --pivot-dimension=source and then the logic
+        # can be generalized to handle that...
+        parser.add_argument('--other-dimensions',
+                            default='ga:source',
+                            dest='otherdimensions',
+                            help='Other data dimensions to include when querying for specified metrics, such as page path')
+
+        parser.add_argument('--other-filters',
+                            default='ga:source=~.*facebook.com|twitter.com|(direct)|^google$',
+                            dest='otherfilters',
+                            help='Filters for the other data dimensions')
+
+    def handle(self, *args, **options):
+        # Build the service to communicate with the Google Analytics API
+        scopes = ['https://www.googleapis.com/auth/analytics.readonly']
+        key_file_location = utils.get_key_file_location()
+        service_account_email = utils.get_service_account_email()
+        service = utils.get_service(
+            'analytics',
+            'v3',
+            scopes,
+            key_file_location,
+            service_account_email
+        )
+
+        # Get the profile to use with the service
+        profile_id = utils.get_first_profile_id(service)
+
+        # Get parameters
+        start_date = options['startdate']
+        end_date = options['enddate']
+        dimensions = options['dimensions']
+        metrics = options['metrics']
+        other_dimensions = options['otherdimensions']
+        other_filters = options['otherfilters']
+
+        # Get data about the pages from database
+        db_statistics_by_page_url = self._get_statistics_from_database(start_date, end_date)
+
+        # Get data with statistics from Google Analytics
+        ga_statistics, other_ga_stats = self._get_statistics_from_google_analytics(service, profile_id,
+                                                                                   start_date, end_date,
+                                                                                   dimensions, metrics,
+                                                                                   other_dimensions, other_filters)
+
+        # Merge the statistics from the different data sources and print them
+        self._print_statistics(db_statistics_by_page_url, ga_statistics, other_ga_stats)
```

Comment:
Design thought: Clean architecture would suggest that print should be it own function so you can test the merging with just data.

https://8thlight.com/blog/uncle-bob/2012/08/13/the-clean-architecture.html